### ISSUE TITLE
The loss function should ignore tokens whose index is set to - 100

### ISSUE THREAD

**Issue by Silypie** [huggingface/transformers#14694]
Loss function in `EncoderDecoderModel` as well as `BartForConditionalGeneration` didn't ignore special tokens:
https://github.com/huggingface/transformers/blob/e9800122a6f6a68aee7dff347c8f4a6d28e345a2/src/transformers/models/encoder_decoder/modeling_encoder_decoder.py#L529

Does it need to be like the loss function in `T5ForConditionalGeneration`:
https://github.com/huggingface/transformers/blob/e9800122a6f6a68aee7dff347c8f4a6d28e345a2/src/transformers/models/t5/modeling_t5.py#L1644

**Comment by NielsRogge** [comment]
No, the ignore_index is set to -100 by default as can be seen in PyTorch' [docs](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html).

**Comment by Silypie** [comment]
> No, the ignore_index is set to -100 by default as can be seen in PyTorch' [docs](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html).

I got it, thank you very much!
